### PR TITLE
Feature/centralize dom flush

### PR DESCRIPTION
**Pull request by bigopon**
# Pull Request

## 📖 Description

Centralize flush changes queue-ing to `IDOM` so we have even nicer perf and remove the pressure on the GC, since we no longer need to constantly creating tasks. In 1kcomponents demo, it seems without any optimization, we are on par with Inferno, worse case scenario is 1ms slower. With a few more optimization, we should be able to break through.

### 🎫 Issues

Nil

## 👩‍💻 Reviewer Notes

Nil

## ⏭ Next Steps

Improve renderer/ binding eval/connect to make start up/binding runtime faster.

### CODE REVIEW

File: packages/runtime/src/binding/property-binding.ts [hunk 1]
```diff
@@ -76,6 +73,7 @@ export class PropertyBinding implements IPartialConnectableBinding {
     connectable.assignIdTo(this);
     this.$lifecycle = locator.get(ILifecycle);
     this.$scheduler = locator.get(IScheduler);
+    this.dom = locator.get(IDOM);
```
Comment (bigopon) [inline packages/runtime/src/binding/property-binding.ts:76]: this will be passed from the renderer

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://53872-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://53878-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54087-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54193-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54235-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54358-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54381-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54431-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54611-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54621-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by bigopon** [comment]
There've been a lot of changes, and the main idea of this PR can easily be incorporated anytime, just not now.

**Pull request closed without merging**